### ISSUE TITLE
SpyderKernelApp] WARNING | No such comm: e7092e33855511ec82fcf40669f143a1

### ISSUE THREAD

**Issue by yeslingo** [spyder-ide/spyder#17294]
### What steps will reproduce the problem?

not sure how to recreate the issue thought, i ran a bunch of code...

### Traceback
```python-traceback
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 176, in comm_msg
    comm.handle_msg(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 264, in handle_msg
    return self._msg_callback(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 330, in _comm_message
    self._message_handlers[spyder_msg_type](
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 338, in _handle_remote_call
    self.on_incoming_call(msg_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 156, in on_incoming_call
    return super(KernelComm, self).on_incoming_call(call_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 402, in on_incoming_call
    self._set_pickle_protocol(call_dict["pickle_highest_protocol"])
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 251, in _set_pickle_protocol
    self._comms[self.calling_comm_id]['pickle_protocol'] = protocol
KeyError: 'd098be98855611eca6b2f40669f143a1'
ERROR:traitlets:Exception in comm_msg for d098be98855611eca6b2f40669f143a1
Traceback (most recent call last):
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 176, in comm_msg
    comm.handle_msg(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 264, in handle_msg
    return self._msg_callback(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 330, in _comm_message
    self._message_handlers[spyder_msg_type](
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 338, in _handle_remote_call
    self.on_incoming_call(msg_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 156, in on_incoming_call
    return super(KernelComm, self).on_incoming_call(call_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 402, in on_incoming_call
    self._set_pickle_protocol(call_dict["pickle_highest_protocol"])
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 251, in _set_pickle_protocol
    self._comms[self.calling_comm_id]['pickle_protocol'] = protocol
KeyError: 'd098be98855611eca6b2f40669f143a1'
ERROR:traitlets:Exception in comm_msg for d098be98855611eca6b2f40669f143a1
Traceback (most recent call last):
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 176, in comm_msg
    comm.handle_msg(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 264, in handle_msg
    return self._msg_callback(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 330, in _comm_message
    self._message_handlers[spyder_msg_type](
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 338, in _handle_remote_call
    self.on_incoming_call(msg_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 156, in on_incoming_call
    return super(KernelComm, self).on_incoming_call(call_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 402, in on_incoming_call
    self._set_pickle_protocol(call_dict["pickle_highest_protocol"])
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 251, in _set_pickle_protocol
    self._comms[self.calling_comm_id]['pickle_protocol'] = protocol
KeyError: 'd098be98855611eca6b2f40669f143a1'
ERROR:traitlets:Exception in comm_msg for d098be98855611eca6b2f40669f143a1
Traceback (most recent call last):
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 176, in comm_msg
    comm.handle_msg(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\qtconsole\comms.py", line 264, in handle_msg
    return self._msg_callback(msg)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 330, in _comm_message
    self._message_handlers[spyder_msg_type](
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 338, in _handle_remote_call
    self.on_incoming_call(msg_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 156, in on_incoming_call
    return super(KernelComm, self).on_incoming_call(call_dict)
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 402, in on_incoming_call
    self._set_pickle_protocol(call_dict["pickle_highest_protocol"])
  File "C:\Users\Lingyan\anaconda3\envs\Exper\lib\site-packages\spyder_kernels\comms\commbase.py", line 251, in _set_pickle_protocol
    self._comms[self.calling_comm_id]['pickle_protocol'] = protocol
KeyError: 'd098be98855611eca6b2f40669f143a1'
```

## Versions

* Spyder version: 4.2.5 
* Python version: 3.8.12
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  2.0.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.0.2 (OK)
IPython >=7.6.0                 :  8.0.1 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  4.4.0 (OK)
keyring >=17.0.0                :  23.4.0 (OK)
nbconvert >=4.0                 :  6.4.1 (OK)
numpydoc >=0.6.0                :  1.2 (OK)
paramiko >=2.4.0                :  2.9.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.9.0 (OK)
pygments >=2.0                  :  2.11.2 (OK)
pylint >=1.0                    :  2.12.2 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8;<3.0           :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.1.1 (OK)
qtconsole >=5.0.3               :  5.2.2 (OK)
qtpy >=1.5.0                    :  2.0.1 (OK)
rtree >=0.8.3                   :  0.9.7 (OK)
setuptools >=39.0.0             :  60.7.1 (OK)
sphinx >=0.6.6                  :  4.4.0 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.2 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  22.3.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  None (NOK)
numpy >=1.7                     :  1.22.1 (OK)
pandas >=1.1.1                  :  1.4.0 (OK)
scipy >=0.17.0                  :  None (NOK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @yeslingo, thanks for reporting. This error is already fixed in our latest version (5.2.2).

Since it's still not available in Anaconda, you have two options:

1. Use our Windows installer, which you can find [here](https://github.com/spyder-ide/spyder/releases).
2. Create a new environment with conda-forge packages. For that, please close Spyder, open the Anaconda Prompt and run the following commands there:

        conda create -n spyder-cf -c conda-forge spyder
        conda activate spyder-cf
        spyder

After that, you can connect Spyder to your base environment by following these instructions:

http://docs.spyder-ide.org/current/faq.html#using-existing-environment

**Comment by yeslingo** [comment]
thank you
